hyper/transport/grpc/restore.go

```diff
@@ -20,14 +19,19 @@ func (h *HyperServer) DesktopRestore(ctx context.Context, req *proto.DesktopRest
 		return nil, err
 	}
 
+	if _, err := os.Stat(req.Path); err != nil {
+		if os.IsNotExist(err) {
+			return nil, status.Error(codes.NotFound, err.Error())
+		}
+		return nil, status.Error(codes.Unknown, err.Error())
+	}
+
```

Comment:
Is this required? Or is there a specific libvirt error that we could get?

Reply:
We need to check that a file with the memory dump exists so it can be restored. Maybe we can try to catch it from libvirt itseld when trying to restore domain.

Reply:
I modified how it handles errors

Reply:
It's still there

Reply:
Yes, it can't be handled by libvirt. DomainRestore returns 0 in case of success and -1 in case of failure.